forkmanager/fork.go

```diff
@@ -41,16 +53,20 @@ func ForkManagerInit(factory func(*chain.Forks) error, forks *chain.Forks) error
 	fm.Clear()
 
 	// register initial fork
-	fm.RegisterFork(InitialFork)
+	fm.RegisterFork(InitialFork, initialParams)
 
 	// Register forks
-	for name := range *forks {
+	for name, f := range *forks {
 		// check if fork is not supported by current edge version
 		if _, found := (*chain.AllForksEnabled)[name]; !found {
 			return fmt.Errorf("fork is not available: %s", name)
 		}
 
-		fm.RegisterFork(name)
+		if f != nil {
```

Comment:
When fork would be nil? Is it in case somebody manually deletes fork from the genesis?

Reply:
yup, exactly. but maybe we can make fork as value not pointer?

Reply:
Is it a valid use case in the first place? In case it isn't, maybe returning an error would be better. WDYT?